plugins/woocommerce/src/Admin/API/LaunchYourStore.php

```diff
@@ -219,4 +231,35 @@ public function update_survey_status( \WP_REST_Request $request ) {
 	public function has_survey_completed() {
 		return new \WP_REST_Response( get_option( 'woocommerce_admin_launch_your_store_survey_completed', 'no' ) );
 	}
+
+	/**
+	 * Check if we're still displaying the coming soon page.
+	 *
+	 * @return \WP_REST_Response
+	 */
+	public function is_coming_soon_page_shown() {
+		$store_pages_only = get_option( 'woocommerce_store_pages_only', 'no' ) === 'yes';
+		if ( $store_pages_only ) {
+			$url = get_permalink( wc_get_page_id( 'shop' ) );
+		} else {
+			$url = site_url();
+		}
```

Comment:
Maybe we can always check shop page? since it still works even if store pages only is disabled.

There might be an edge case where users unset the shop page (even though it's weird, users launch the store without the shop page 😅 ) and cause $url be empty, so we probably need to handle that case.

Reply:
@chihsuan Good call!